irods/manager/metadata_manager.py

```diff
@@ -96,8 +96,9 @@ def meta_opts(row):
 
     def add(self, model_cls, path, meta, **opts):
         # Avoid sending request with empty argument(s)
-        if not(len(path) and len(meta.name) and len(meta.value)):
-            raise ValueError('Empty value in ' + repr(meta))
+        if type(meta.name) == 'str' and type(meta.value) == 'str':
```

Comment:
Not sure about this line. If an integer is passed for either `.name` or `.value`, the checks are skipped entirely.

A approach is to enforce that the types of those members are strings and throw an error if they aren't.

Reply:
Also,I believe type(n) is never equal to 'str'. I think  the global str type was what's meant?

Reply:
Correct. I had quickly tested this before pushing, my bad! Even the following works:

`if type(meta.name) == 'something' and type(meta.value) == 'something':`

even with this line fix, when an integer is passed with name or value, I getting the expected result. I am investigating.

Reply:
@d-w-moore , I guess the solution is to delete L98-100 completely (seems no need to check length)? Because then it is complaining correctly in case both passing an empty string and non-string name value. I am now checking one more thing.

Reply:
So basically this is already checking length for an attempt to enter an empty string:
https://github.com/irods/python-irodsclient/blob/main/irods/message/__init__.py#L701-L715

here is my test after I remove lines in the add method:
```
s = make_session()
target = s.collections.get("/u0137480/home/rods/colA")
meta_set = iRODSMeta("", 2, "units")
meta_add = iRODSMeta(3, "", "units")
print("set exception:")
try:
    target.metadata.set(meta_set)
except Exception as e:
    print(e)
print("\nadd exception:")
try:
    target.metadata.add(meta_add)
except Exception as e:
    print(e)
```
```
set exception:
AVU attribute ('') is zero-length.

add exception:
AVU attribute (3) has incorrect type. AVU fields must be strings, except for units, which could be None.

```

Reply:
Lastly, changing ValuError here https://github.com/irods/python-irodsclient/blob/main/irods/test/meta_test.py#L260 with `with self.assertRaisesRegexp(Bad_AVU_Field,'zero-length'):` the unittest passed.



```
[irods@dev-u0137480 irods]$ python3 -m unittest irods.test.meta_test
../usr/local/lib/python3.9/site-packages/python_irodsclient-2.0.0-py3.9.egg/irods/test/meta_test.py:261: DeprecationWarning: Please use assertRaisesRegex instead.
  with self.assertRaisesRegexp(Bad_AVU_Field,'zero-length'):#("irods.message.Bad_AVU_Field: AVU value ('') is zero-length."):
...........s..s..s
----------------------------------------------------------------------
Ran 20 tests in 13.000s

OK (skipped=3)
```

Reply:
Looking again, I think [this  class's constructor ](https://github.com/irods/python-irodsclient/blob/04baaa8c0d3591276f335bca75648b5b5c392acb/irods/message/__init__.py#L692) does all the appropriate checks, so probably we should just delete the checks being done [here](https://github.com/irods/python-irodsclient/blame/04baaa8c0d3591276f335bca75648b5b5c392acb/irods/manager/metadata_manager.py#L99).

That might be the solution that gets us consistency in error messages.

Reply:
Please test this theory - get a test written that exercises this codepath and assert the error messages.

Reply:
a2f56332 was implemented for release 1.1.7 and performed all such checks within the `MetadataRequest` constructor.  It should probably have deleted [these checks](https://github.com/irods/python-irodsclient/blame/04baaa8c0d3591276f335bca75648b5b5c392acb/irods/manager/metadata_manager.py#L99) that 3bfe3d11 introduced for v0.6.0 (the same mentioned in the comment just above)

Reply:
> Please test this theory - get a test written that exercises this codepath and assert the error messages.

Yes, sounds good.  Are we considering this for the current release?

Reply:
I see. Thanks